src/main/java/io/nats/client/impl/NatsConnection.java

```diff
@@ -283,19 +285,13 @@ void reconnect() throws InterruptedException {
 
                 connectError.set(""); // reset on each loop
 
-                if (server.equals(lastServer)) {
-                    this.reconnectWaiter = new CompletableFuture<>();
-                    waitForReconnectTimeout();
-                }
-
                 if (isDisconnectingOrClosed() || this.isClosing()) {
                     break;
                 }
 
                 updateStatus(Status.RECONNECTING);
 
                 tryToConnect(server, System.nanoTime());
-                lastServer = server;

```

Comment:
Do I read this right, removing this line will bring the client in lie with other clients and fix #290

Reply:
It should, yes.